admin/tabs/roles/details/members/RoleMembersModel.ts

```diff
@@ -0,0 +1,205 @@
+                    type: RoleMembersModel.types.ROLE
+                }))
+            ]);
+        } else {
+            this.gridModel.loadData(
+                role.members.map(it => ({...it, type: RoleMembersModel.types[it.type]}))
+            );
+        }
+    }
+
+    private createGridModel(): GridModel {
+        const {types} = RoleMembersModel,
+            {showEffective} = this.props;
+        return new GridModel({
+            store: {
+                fields: [
+                    {name: 'name', type: 'string'},
+                    {name: 'type', type: 'string'},
+                    {name: 'sources', displayName: 'Assigned Via', type: 'json'}
+                ],
+                idSpec: data => `${this.selectedRole.name}:${data.type}:${data.name}`
+            },
+            emptyText: 'This role has no members.',
+            filterModel: true,
+            groupBy: 'type',
+            headerMenuDisplay: 'hover',
+            sortBy: 'name',
+            onRowDoubleClicked: ({data: record}) => {
+                if (!record) return;
+                const {type, name} = record.data;
+                if (type === types.ROLE) this.rolesModel.selectRole(name);
+            },
+            groupRowRenderer: ({value}) => {
+                switch (value) {
+                    case types.USER:
+                        return hframe({
+                            className: 'group-row-renderer',
+                            items: [Icon.user(), 'Users']
+                        });
+                    case types.DIRECTORY_GROUP:
+                        return hframe({
+                            className: 'group-row-renderer',
+                            items: [Icon.users(), 'Directory Groups']
+                        });
+                    case types.ROLE:
+                        return hframe({
+                            className: 'group-row-renderer',
+                            items: [Icon.idBadge(), 'Roles']
+                        });
+                }
+            },
+            colDefaults: {
+                filterable: true
+            },
+            columns: [
+                {field: 'name'},
+                {field: 'type', hidden: true},
+                {
+                    field: 'sources',
+                    flex: true,
+                    renderer: (sources: Array<{role: string; directoryGroup?: string}>) =>
+                        hbox({
+                            className: 'roles-renderer',
+                            items: uniqBy(
+                                sources.map(({role, directoryGroup}) => {
+                                    const isThisRole = role === this.selectedRole.name;
+                                    return {
+                                        className: 'roles-renderer__role',
+                                        intent: isThisRole ? null : 'primary',
```

Comment:
The use of the primary intent here makes it look a bit like these are links to jump to that role - maybe they should be?

Reply:
Had that thought too, will implement